packages/transaction-controller/src/TransactionController.ts

```diff
@@ -2439,8 +2445,8 @@ export class TransactionController extends BaseController<
 
   private async updateGasProperties(transactionMeta: TransactionMeta) {
     const isEIP1559Compatible =
-      (await this.getEIP1559Compatibility(transactionMeta.networkClientId)) &&
-      transactionMeta.txParams.type !== TransactionEnvelopeType.legacy;
+      transactionMeta.txParams.type !== TransactionEnvelopeType.legacy &&
```

Comment:
Swapped operands so that compatibility can be determined synchronously by tx type in some cases.